src/app/component/user/components/profile/shopping-list/shopping-list.component.html

```diff
@@ -1,11 +1,24 @@
-<div class="shopping-list-content">
-    <div class="header-position">
-        <h2 class="header">Shopping list <span>{{ shoppingList.length }}</span></h2>
-        <a class="btn-see-all">See all</a>
+<div class="header-position shopping-list-content">
+    <h2 class="header">Shopping list <span>{{ shoppingList?.length }}</span></h2>
+    <a class="btn-see-all">See all</a>
+</div>
+<div class="shopping-list-content" *ngIf="shoppingList; else loading;">
+    <div *ngIf="shoppingList.length; else empty;">
+        <ul class="shopping-list">
+            <li class="item" *ngFor="let shoppingItem of shoppingList">{{ shoppingItem.text }}
+                <mat-checkbox [checked]='shoppingItem.status === "DONE"' class="checkbox" color="primary"
+                    (click)="toggleDone(shoppingItem)"></mat-checkbox>
+            </li>
+        </ul>
     </div>
-    <ul class="shopping-list">
-        <li class="item" *ngFor="let spoppingItem of getShoppingList()">{{ spoppingItem }}
-            <mat-checkbox class="checkbox" color="primary"></mat-checkbox>
-        </li>
-    </ul>
 </div>
+<ng-template #empty>
+    <p class="noItems">You don't have any items in your Shopping List</p>
+</ng-template>
+<ng-template #loading>
+    <mat-spinner 
+        color="grey" 
```

Comment:
Bad alignment: first attribute in the same row, where the tag is, the others should be in separate row, under the first. Check any .html file in the project to understand it better

Reply:
done